src/database/hooks/useSaveAnonChatHook.ts

```diff
@@ -53,65 +85,38 @@ const useSaveAnonChatHook = () => {
 
     initAnonymousChat.message.cid = initAnonymousChat.message.cid.replace('messaging:', '');
 
-    const chat = ChatSchema.fromInitAnonymousChatAPI(initAnonymousChat, status);
-    await chat.save(localDb);
+    try {
+      await helperSaveChannel(initAnonymousChat, type);
+    } catch (e) {
+      console.log(e);
+    }
 
     try {
+      const chat = ChatSchema.fromInitAnonymousChatAPI(initAnonymousChat, status);
+      await chat.save(localDb);
       const members = originalMembers || object?.members;
-      members?.forEach((member) => {
-        const saveUserAndChannelListMember = async () => {
-          try {
-            const userMember = UserSchema.fromInitAnonymousChatAPI(
-              member,
-              initAnonymousChat?.message?.cid
-            );
-
-            await userMember.saveOrUpdateIfExists(localDb);
-          } catch (e) {
-            console.log('error saveChatFromOtherProfile userMember', e);
-          }
-          try {
-            const memberSchema = ChannelListMemberSchema.fromInitAnonymousChatAPI(
-              initAnonymousChat?.message?.cid,
-              uuid(),
-              member
-            );
-            memberSchema.saveIfNotExist(localDb);
-          } catch (e) {
-            console.log('error saveChatFromOtherProfile memberSchema', e);
-          }
-        };
-
-        Promise.all([saveUserAndChannelListMember()]);
+      const promises: Promise<void>[] = members?.map((member) => {
+        return saveUserAndChannelListMember(initAnonymousChat.message.cid, member);
       });
+
+      await Promise.all(promises);
     } catch (e) {
-      console.log('error saveChatFromOtherProfile');
       console.log(e);
+    } finally {
+      refresh('channelList');
+      refresh('chat');
+      if (withNavigate) {
+        helperGoToAnonymousChat(object, GROUP_INFO);
+      }
     }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, error handling could be improved in the `saveChatFromOtherProfile` function. If an error occurs during the execution of `helperSaveChannel` or while saving the chat and its members, it's logged but not propagated further. Consider throwing the error after logging it or returning a rejected Promise.

```diff
    try {
      await helperSaveChannel(initAnonymousChat, type);
    } catch (e) {
      console.log(e);
+      throw e;
    }
 
     try {
      const chat = ChatSchema.fromInitAnonymousChatAPI(initAnonymousChat, status);
      await chat.save(localDb);
       const members = originalMembers || object?.members;
      const promises: Promise<void>[] = members?.map((member) => {
        return saveUserAndChannelListMember(initAnonymousChat.message.cid, member);
       });

      await Promise.all(promises);
     } catch (e) {
       console.log(e);
+      throw e;
    } finally {
      refresh('channelList');
      refresh('chat');
      if (withNavigate) {
        helperGoToAnonymousChat(object, GROUP_INFO);
      }
     }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->